examples/tags-in-searchbox/app.tsx

```diff
@@ -0,0 +1,365 @@
+/** @jsx h */
+import {
+  autocomplete,
+  AutocompleteComponents,
+  getAlgoliaResults,
+  getAlgoliaFacets,
+} from '@algolia/autocomplete-js';
+import {
+  AutocompleteInsightsApi,
+  createAlgoliaInsightsPlugin,
+} from '@algolia/autocomplete-plugin-algolia-insights';
+import { createTagsPlugin, Tag } from '@algolia/autocomplete-plugin-tags';
+import algoliasearch from 'algoliasearch';
+import { h, Fragment, render } from 'preact';
+import groupBy from 'ramda/src/groupBy';
+import insightsClient from 'search-insights';
+
+import '@algolia/autocomplete-theme-classic';
+
+import { ProductHit, TagExtraData } from './types';
+
+const appId = 'latency';
+const apiKey = '6be0576ff61c053d5f9a3225e2a90f76';
+const searchClient = algoliasearch(appId, apiKey);
+
+// @ts-expect-error type error in search-insights
+insightsClient('init', { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({ insightsClient });
+
+const tagsPlugin = createTagsPlugin<TagExtraData>({
+  getTagsSubscribers() {
+    return [
+      {
+        sourceId: 'brands',
+        getTag({ item }) {
+          return item;
+        },
+      },
+      {
+        sourceId: 'categories',
+        getTag({ item }) {
+          return item;
+        },
+      },
+    ];
+  },
+  transformSource() {
+    return undefined;
+  },
+  onChange({ tags }) {
+    requestAnimationFrame(() => {
```

Comment:
why is this in an animation frame?

Reply:
This ensures the DOM update is scheduled after the library has re-rendered the searchbox.

Reply:
how do you know there's not another update already in progress at that point? is there a chance of tearing here?

Reply:
Actually scratch that—after digging a bit, it seems that we never fully re-render the whole searchbox, [only a set of select children](https://github.com/algolia/autocomplete/blob/c3aeb9f817dda921f77a40a9b9c9010557cfd1c0/packages/autocomplete-js/src/render.tsx#L33-L60). The input wrapper prefix (which is where we inject tags in this demo) is created once then we never touch it.

So in this case, `rAF` is not useful for rendering order, just for pushing these manual DOM upates in macrotask queue.